burr/integrations/opentelemetry.py

```diff
@@ -66,21 +67,21 @@ def get_cached_span(span_id: int) -> Optional[FullSpanContext]:
 tracker_context = ContextVar[Optional[SyncTrackingClient]]("tracker_context", default=None)
 
 
-def _is_heterogeneous_array(value: list):
+def _is_heterogeneous_sequence(value: Sequence):
     if len(value) == 0:
         return True
     first_type = type(value[0])
-    return any([not isinstance(val, first_type) for val in value])
+    return all([isinstance(val, first_type) for val in value])
```

Comment:
doesn't this need the negation? e.g. it's homogeneous with `all()` hetro if not.

Reply:
yeah the function is named poorly